language/src/regent/codegen.t

```diff
@@ -8018,6 +8018,42 @@ function codegen.expr_import_partition(cx, node)
     partition_type)
 end
 
+function codegen.expr_import_cross_product(cx, node)
+  local partitions = node.partitions:map(function(p) return codegen.expr(cx, p):read(cx) end)
+  local colors = codegen.expr(cx, node.colors):read(cx)
+  local value = codegen.expr(cx, node.value):read(cx)
+  local expr_type = std.as_read(node.expr_type)
+
+  local actions = quote
+    [colors.actions];
+    [value.actions];
+  end
+
+  for _, p in pairs(partitions) do
+    actions = quote
+      [actions];
+      [p.actions];
+    end
+  end
```

Comment:
We should be careful to codegen this prior to `colors.actions`, so that we're not reversing the order of operations of the arguments.